scene/gui/rich_text_label.cpp

```diff
@@ -5726,19 +5726,89 @@ void RichTextLabel::set_text_direction(Control::TextDirection p_text_direction)
 
 	if (text_direction != p_text_direction) {
 		text_direction = p_text_direction;
-		main->first_invalid_line.store(0); //invalidate ALL
-		_validate_line_caches();
+		if (!text.is_empty()) {
+			_apply_translation();
+		} else {
+			main->first_invalid_line.store(0); //invalidate ALL
```

Comment:
```suggestion
			main->first_invalid_line.store(0); // Invalidate all lines.
```
Same for all other occurrences.